python/ray/_raylet.pyx

```diff
@@ -722,6 +723,30 @@ cdef void delete_spilled_objects_handler(
                 job_id=None)
 
 
+cdef void unhandled_exception_handler(const CRayObject& error) nogil:
+
+    with gil:
+        worker = ray.worker.global_worker
+        data = None
+        metadata = None
+        if error.HasData():
+            data = Buffer.make(error.GetData())
+        if error.HasMetadata():
+            metadata = Buffer.make(error.GetMetadata()).to_pybytes()
+        results = worker.deserialize_objects(
+            [(data, metadata)], [ray.ObjectRef.nil()])
+        print("DESERIALIZATION DONE")
```

Comment:
@edoakes any idea why deserialization would succeed (deserialize_objects prints "DESER RESULT..." ), but then we never get to this second print?

The output I get is
```
DESERIALIZATION RESULT [RayTaskError('__main__.f', 'Traceback (most recent call last):\n  File "python/ray/_raylet.pyx", line 487, in ray._raylet.execute_task\n  File "test.py", line 7, in f\n    raise ValueError("hi")\nValueError: hi\n', <class 'ValueError'>)]
terminate called without an active exception
zsh: abort      python test.py
```

Reply:
I have no idea why you're not hitting the 2nd print, but the only time I've seen this error message is when a c++ thread goes out of scope without joining it or detaching it. If this is being called on a callback from the core worker there might be something weird happening with the lifetime management in c++